File: src/test/java/cube8540/oauth/authentication/credentials/oauth/client/infra/rule/DefaultClientNameValidationRuleTest.java
package cube8540.oauth.authentication.credentials.oauth.client.infra.rule;

import cube8540.oauth.authentication.credentials.oauth.client.domain.OAuth2Client;
import org.junit.jupiter.api.BeforeEach;
import org.junit.jupiter.api.DisplayName;
import org.junit.jupiter.api.Test;

import static org.junit.jupiter.api.Assertions.assertFalse;
import static org.mockito.Mockito.mock;
import static org.mockito.Mockito.when;

@DisplayName("기본 클라이언트명 유효성 검사 테스트")
class DefaultClientNameValidationRuleTest {

    private DefaultClientNameValidationRule rule;

    @BeforeEach
    void setup() {
        this.rule = new DefaultClientNameValidationRule();
    }

    @Test
    @DisplayName("클라이언트명이 null 일시 유효성 검사 결과는 false 가 반환되어야 한다.")
    void clientNameIsNullValidationShouldReturnsFalse() {
        OAuth2Client client = mock(OAuth2Client.class);

        when(client.getClientName()).thenReturn(null);

        assertFalse(rule.isValid(client));
    }
}

File: src/test/kotlin/cube8540/oauth/authentication/oauth/client/infra/DefaultClientNameValidationRuleTest.kt
package cube8540.oauth.authentication.oauth.client.infra

import cube8540.oauth.authentication.oauth.client.domain.OAuth2Client
import io.mockk.every
import io.mockk.mockk
import org.assertj.core.api.Assertions.assertThat
import org.junit.jupiter.api.Test

class DefaultClientNameValidationRuleTest {

    private val rule = DefaultClientNameValidationRule()

    @Test
    fun `client name must not be null`() {
        val client: OAuth2Client = mockk {
            every { clientName } returns null
        }

        val result = rule.isValid(client)
        assertThat(result).isFalse
    }
}